frontend/src/metabase/querying/components/FilterModal/SegmentFilterEditor/SegmentFilterEditor.tsx

```diff
@@ -0,0 +1,55 @@
+import { useMemo } from "react";
```

Comment:
I removed a segment and went back to the query builder - the removed segment was still available in the filter modal (until I refreshed a page).

In `master` this does not happen.